lib/IRGen/IRGenSIL.cpp

```diff
@@ -1916,6 +1918,8 @@ static void abortOnGraphOp(IRGenFunction &IGF, llvm::StringRef errMessage) {
 }
 
 // Create and return a (i8*-typed) address value to a constant string.
+// TODO: As a code size optimization, if we have created this global string
+// before, we can avoid creating a new one.
```

Comment:
I think that would require defining this as a method in `IRGenSILFunction::`. Would it make sense to do it now? IIUC we may be generate new strings super often, since each op has a device name attribute.